shared/src/address/payload.rs

```diff
@@ -3,9 +3,63 @@
 
 use std::convert::TryInto;
 use std::hash::Hash;
+use std::ops::Deref;
 use std::u64;
 
-use super::{from_leb_bytes, to_leb_bytes, Error, Protocol, BLS_PUB_LEN, PAYLOAD_HASH_LEN};
+use super::{
+    from_leb_bytes, to_leb_bytes, Error, Protocol, BLS_PUB_LEN, MAX_SUBADDRESS_LEN,
+    PAYLOAD_HASH_LEN,
+};
+use crate::ActorID;
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub struct Subaddress {
```

Comment:
Please bikeshed the name.

Reply:
`ManagedAddress`?
 
 Really new address are aliases\delegated redirects to an actor, but those names have added baggage that doesn't apply.